pkg/controller/signup.go

```diff
@@ -66,19 +66,37 @@ func (s *Signup) GetHandler(ctx *gin.Context) {
 // VerifyCodeHandler validates the phone verification code passed in by the user
 func (s *Signup) VerifyCodeHandler(ctx *gin.Context) {
 	code := ctx.Param("code")
+	if code == "" {
+		log.Error(ctx, nil, "no code provided in request")
+		ctx.AbortWithStatus(http.StatusNotFound)
+	}
 
 	userID := ctx.GetString(context.SubKey)
 	signupResource, err := s.signupService.GetUserSignup(userID)
 	if err != nil {
 		log.Error(ctx, err, "error getting UserSignup resource")
 		errors.AbortWithError(ctx, http.StatusInternalServerError, err, "error getting UserSignup resource")
+		return
 	}
+
 	if signupResource == nil {
 		log.Errorf(ctx, nil, "UserSignup resource for userID: %s resource not found", userID)
 		ctx.AbortWithStatus(http.StatusNotFound)
+		return
 	}
 
 	err = s.verificationService.VerifyCode(ctx, signupResource, code)
+
+	// Regardless of whether the VerifyCode() call returns an error or not, we need to update the UserSignup instance
+	// as its state can be updated even in the case of an error.  This may result in the slight possibility that any
+	// errors returned by VerifyCode() are suppressed, as error handling for the UserSignup update is given precedence.
```

Comment:
Let's at least log the error returned by VerifyCode() if the update failed too.

Reply:
Done.